malariagen_data/anopheles.py

```diff
@@ -6358,20 +6361,52 @@ def plot_samples_interactive_map(
             df_location_aggs, on=location_composite_key, validate="one_to_one"
         )
 
-        debug("create a map")
-        if basemap is None:
-            basemap = ipyleaflet.basemaps.Esri.WorldImagery
+        debug("create basemap")
+        basemap_abbrevs = {
+            "mapnik": ipyleaflet.basemaps.OpenStreetMap.Mapnik,
+            "natgeoworldmap": ipyleaflet.basemaps.Esri.NatGeoWorldMap,
+            "opentopomap": ipyleaflet.basemaps.OpenTopoMap,
+            "positron": ipyleaflet.basemaps.CartoDB.Positron,
+            "satellite": ipyleaflet.basemaps.Gaode.Satellite,
+            "terrain": ipyleaflet.basemaps.Stamen.Terrain,
+            "watercolor": ipyleaflet.basemaps.Stamen.Watercolor,
+            "worldimagery": ipyleaflet.basemaps.Esri.WorldImagery,
+            "worldstreetmap": ipyleaflet.basemaps.Esri.WorldStreetMap,
+            "worldtopomap": ipyleaflet.basemaps.Esri.WorldTopoMap,
+        }
+
+        # Get basemap_provider
+        if type(basemap) == str:
```

Comment:
Will do.
I reckon it's because using `isinstance` instead of `type` can check that the object is an instance of `str` as well as other classes that inherit from `str`. In this case, I suppose `basemap` could, in principle, be a string but not actually have type`str`, e.g. when it's a type that inherits from the `str` class.